src/ucp/api/ucp.h

```diff
@@ -3768,19 +3768,17 @@ void ucp_dt_destroy(ucp_datatype_t datatype);
  * When @ref UCP_DATATYPE_ATTR_FIELD_PACKED_SIZE is set in @a field_mask of @a attr,
  * the field @a packed_size is set to the packed size (bytes) of the datatype.
  *
- * @note The size of a generic datatype is available only after it has been used
- *       in a communication operation. If it is queried before,
- *       @ref UCS_ERR_UNSUPPORTED will be returned.
- *
- * @note The size of IOV data cannot be determined from the UCP datatype. If an
- *       IOV @a datatype is passed, @ref UCS_ERR_INVALID_PARAM will be returned.
+ * @note The parameters @a buffer and @a count are ignored for contiguous datatypes.
  *
  * @param [in]  datatype     Datatype object to query.
+ * @param [in]  buffer       Buffer for generic or IOV datatypes (@ref ucp_dt_iov_t).
+ * @param [in]  count        Number of elements in @a buffer.
  * @param [out] attr         Filled with attributes of the datatype.
  *
  * @return Error code as defined by @ref ucs_status_t
  */
-ucs_status_t ucp_dt_query(ucp_datatype_t datatype, ucp_datatype_attr_t *attr);
+ucs_status_t ucp_dt_query(ucp_datatype_t datatype, const void *buffer,
```

Comment:
@shamisp do you think it would be better to make buffer,count parameters optional using a struct?
PS. This API was not released yet

Reply:
@yosefe - I think it is a good idea.

Reply:
@rdietric let's move buffer,count into ucp_datatype_attr_t (with corresponding flags)